src/annotation.js

```diff
@@ -144,33 +144,24 @@ function draw(chart, caller, clip) {
     box = {x: chartArea.left, y: chartArea.top, width: chartArea.width, height: chartArea.height};
   }
 
-  drawElements(ctx, visibleElements, caller, box);
+  drawElements(chart, visibleElements, caller, box);
 
   if (clip) {
     unclipArea(ctx);
   }
-
-  visibleElements.forEach(el => {
-    if (!('drawLabel' in el)) {
-      return;
-    }
-    const label = el.options.label;
-    if (label && label.display && label.content && (label.drawTime || el.options.drawTime) === caller) {
-      el.drawLabel(ctx, chartArea);
-    }
-  });
 }
 
-function drawElements(ctx, elements, caller, area) {
+function drawElements(chart, elements, caller, area) {
   for (const el of elements) {
     if (el.options.drawTime === caller) {
-      el.draw(ctx);
+      el.draw(chart.ctx);
     }
     if (el.elements && el.elements.length) {
       const box = 'getBoundingBox' in el ? el.getBoundingBox() : area;
+      const labelIsVisible = 'labelIsVisible' in el ? el.labelIsVisible(false, chart.chartArea) : true;
```

Comment:
Ok, did some tinkering and created a branch that is based on 630 and has some further edits compared to this, could you take a look if you agree with anything:
https://github.com/chartjs/chartjs-plugin-annotation/compare/master...kurkle:subelementLineLabel?expand=1

Reply:
First feedback about:

```javascript
    if (!inside) {
      options.label.display = false;
    }
```

In this way we are overriding what the user set (maybe a scriptable options). Not sure if it could be an issue.

Reply:
The rest sounds very good!! Thank you! What next? Can we merge your PR in mine or better that I apply all your updtaes in this one?

Reply:
if you agree, I would like to merge the PR 630 and then to submit your PR and merge in this one, in order to be aligned. What do you thinK?

Reply:
I have seen better, that PR is not extending this one, therefore we could close this one and apply yours.

Reply:
removed as in the other PR